src/Layers/xrRender_R2/r3_rendertarget_phase_smap_D.cpp

```diff
@@ -2,51 +2,21 @@
 
 void CRenderTarget::phase_smap_direct(light* L, u32 sub_phase)
 {
-    // Targets
-    u_setrt(nullptr, nullptr, nullptr, rt_smap_depth);
-
-    //	Don't have rect clear for DX11
-    //	TODO: DX9:	Full clear must be faster for the near phase for SLI
-    //	inobody clears this buffer _this_ frame.
-    // Clear
-    // if (SE_SUN_NEAR==sub_phase)			{
-    // optimized clear
-    //	D3DRECT		R;
-    //	R.x1		= L->X.D.minX;
-    //	R.x2		= L->X.D.maxX;
-    //	R.y1		= L->X.D.minY;
-    //	R.y2		= L->X.D.maxY;
-    //	CHK_DX							(HW.pDevice->Clear( 1L, &R,	  D3DCLEAR_ZBUFFER,	0xFFFFFFFF, 1.0f, 0L));
-    //} else
+    if (sub_phase == SE_SUN_RAIN_SMAP)
     {
-        // full-clear
-        RCache.ClearZB(rt_smap_depth, 1.0f);
+        u_setrt(nullptr, nullptr, nullptr, rt_smap_rain);
+        RCache.ClearZB(rt_smap_rain, 1.0f);
+        RCache.SetViewport({0, 0, rt_smap_rain->dwWidth, rt_smap_rain->dwHeight, 0.0, 1.0});
     }
-
-    //	Prepare viewport for shadow map rendering
-    if (sub_phase != SE_SUN_RAIN_SMAP)
-        RImplementation.rmNormal();
     else
     {
-        RCache.SetViewport({ (float)L->X.D.minX, (float)L->X.D.minY,
-            (float)(L->X.D.maxX - L->X.D.minX), (float)(L->X.D.maxY - L->X.D.minY),
-            0.f, 1.f });
+        u_setrt(nullptr, nullptr, nullptr, rt_smap_depth);
+        RCache.ClearZB(rt_smap_depth, 1.0f);
+        RImplementation.rmNormal();
     }
 
     // Stencil	- disable
     RCache.set_Stencil(FALSE);
-
-    //	TODO: DX11:	Implement culling reverse for DX11
-    // Misc		- draw only front/back-faces
-    /*
-    if (SE_SUN_NEAR==sub_phase)			RCache.set_CullMode			( CULL_CCW	);	// near
-    else								{
-        if (RImplementation.o.HW_smap)	RCache.set_CullMode			( CULL_CW	);	// far, reversed
-        else							RCache.set_CullMode			( CULL_CCW	);	// far, front-faces
-    }
-    if (RImplementation.o.HW_smap)		RCache.set_ColorWriteEnable	( FALSE		);
-    else								RCache.set_ColorWriteEnable	( );
-    */
```

Comment:
Ah, you were right when you asked about these comments...
Please, don't delete them...

Reply:
Ah, whatever, if we will introduce render-graph these comments won't be needed at all.